src/pyaro_readers/lcsreader/LCSReader.py

```diff
@@ -0,0 +1,182 @@
+
+        self._len_dataset = len(self._dataset)
+
+    def __len__(self) -> int:
+        return self._len_dataset
+
+    def slice(self, index):
+        return LCSData(self._dataset.filter(index), self._variable)
+
+    @property
+    def altitudes(self):
+        return np.zeros(self._len_dataset)
+
+    @property
+    def start_times(self):
+        return self._dataset["start"].to_numpy()
+
+    @property
+    def end_times(self):
+        return self._dataset["stop"].to_numpy()
+
+    @property
+    def quality(self):
+        return self._dataset["quality"].to_numpy()
+
+    @property
+    def network(self):
+        return self._dataset["network"].to_numpy()
+
+    # @property
+    # def qc(self):
+    #     return self._dataset["qc"].to_numpy()
+
+    def keys(self):
+        return set(self._dataset.columns) - set(["units"])
+
+    @property
+    def latitudes(self):
+        return self._dataset["lat"].to_numpy()
+
+    @property
+    def longitudes(self):
+        return self._dataset["lon"].to_numpy()
+
+    @property
+    def stations(self):
+        return self._dataset["station_name"].to_numpy()
+
+    @property
+    def values(self):
+        return self._dataset["PM25"].to_numpy()
+
+    @property
+    def units(self):
+        return "ug m-3"
+
+    @property
+    def standard_deviations(self):
+        return np.zeros(self._len_dataset)
+
+    @property
+    def flags(self):
+        return self._dataset["quality"].to_numpy()
+
+    @property
+    def variable(self) -> str:
+        return self._variable
+
+
+class LCSReader(AutoFilterReader):
+
+    read_columns = [
+        "start",
+        "stop",
+        "station_name",
+        "lon",
+        "lat",
+        "PM25",
+        "spread",
+        # "qc",    # Might want to read this at a later point, to give users better control over quality
+        "quality",
+        "network",
+    ]
+
+    def __init__(
+        self,
+        filename: str,
+        min_quality: int = 2,
+        min_spread: int = 3,
+        network: Literal["PA", "SC", "both"] = "both",
+        *,
+        filters,
+    ):
+
+        self._set_filters(filters)
+
+        if network.lower() not in ["pa", "sc", "both"]:
+            raise LCSReaderException(f"Network must be either PA, SC or both")
+
+        if min_spread > 3 or min_spread < 1:
+            raise LCSReaderException(f"min_spread must be in range [1,3]")
+
+        if min_quality > 2 or min_quality < 0:
+            raise LCSReaderException(f"min_spread must be in range [0,2]")
+
+        dataset = pl.scan_parquet(filename).select(self.read_columns)
+
+        if network != "both":
+            dataset = dataset.filter(pl.col("network").eq(network))
+
+        dataset = dataset.filter(pl.col("spread").ge(float(min_spread)))
+        dataset = dataset.filter(pl.col("quality").ge(float(min_quality)))
+
+        self._dataset = dataset.collect()
+
+    def metadata(self) -> dict[str, str]:
+        return {"revision": "0.0.2"}
```

Comment:
Revision should be a datetime or similar to show cache invalidity. Using the timestamp of the parquet file would be a good option for this reader